src/auth/policy/resolver.go

```diff
@@ -1,41 +1,349 @@
 package policy
 
 import (
+	"fmt"
+	"strings"
+
+	"github.com/armon/go-radix"
 	"github.com/consensys/quorum-key-manager/src/auth/types"
 )
 
+var (
+	wildCardSuffix = "*"
+)
+
 type Operation struct {
 	Action string
 	Path   string
 }
 
-type Result struct {
-	isAllowed bool
-	err       error
+type Result interface {
+	Allowed() bool
+	Error() error
+}
+
+// Resolver is responsible to control whether an operation is authorized or not
+// depending on the set of policies attached to the resolver
+type Resolver interface {
+	IsAuthorized(op *Operation) Result
 }
 
-func (res *Result) Allowed() bool {
-	return res.isAllowed
+type statement struct {
+	types.Statement
+	policy *types.Policy
 }
 
-func (res *Result) Error() error {
-	return res.err
+type result struct {
+	op *Operation
+
+	allow *statement
+	deny  *statement
 }
 
-// Resolver is responsible to control whether an operation is authorized or not
-// depending on the set of policies attached to the resolver
-type Resolver struct {
-	// TODO: to be implemented
+func (res *result) Allowed() bool {
+	return res.deny == nil && res.allow != nil
+}
+
+func (res *result) Error() error {
```

Comment:
In which context did you intend to use this errors? This looks more like responsability of the Audit to define which kind of output message it should be added into the auditing logs but for the Store looking to allow or deny and operation this Error() does not seem useful.